docs/docsite/rst/os_guide/windows_performance.rst

```diff
@@ -20,24 +20,50 @@ seconds.
 
 .. code-block:: powershell
 
-    function Optimize-PowershellAssemblies {
-      # NGEN powershell assembly, improves startup time of powershell by 10x
-      $old_path = $env:path
-      try {
-        $env:path = [Runtime.InteropServices.RuntimeEnvironment]::GetRuntimeDirectory()
-        [AppDomain]::CurrentDomain.GetAssemblies() | % {
-          if (! $_.location) {continue}
-          $Name = Split-Path $_.location -leaf
-          if ($Name.startswith("Microsoft.PowerShell.")) {
-            Write-Progress -Activity "Native Image Installation" -Status "$name"
-            ngen install $_.location | % {"`t$_"}
-          }
-        }
-      } finally {
-        $env:path = $old_path
-      }
+    function Optimize-Assemblies {
+        param (
+            [string]$assemblyFilter = "Microsoft.PowerShell.",
+            [string]$activity = "Native Image Installation"
+        )
+        
+        # Save the current $env:path
+        $old_path = $env:path
+        
+        try {
+            # Set $env:path to the runtime directory
+            $env:path = [Runtime.InteropServices.RuntimeEnvironment]::GetRuntimeDirectory()
+            
+            # Get a list of loaded assemblies
+            $assemblies = [AppDomain]::CurrentDomain.GetAssemblies()
+            
+            # Filter assemblies based on the provided filter
+            $filteredAssemblies = $assemblies | Where-Object { $_.Location -match $assemblyFilter }
+            
+            if ($filteredAssemblies.Count -eq 0) {
+                Write-Host "No matching assemblies found for optimization."
+                return
+            }
+            
+            foreach ($assembly in $filteredAssemblies) {
+                # Get the name of the assembly
+                $name = [System.IO.Path]::GetFileName($assembly.Location)
+                
+                # Display progress
+                Write-Progress -Activity $activity -Status "Optimizing $name"
+                
+                # Use Ngen to install the assembly
+                ngen install $assembly.Location | ForEach-Object { "`t$_" }
+            }
+            
+            Write-Host "Optimization complete."
+        } finally {
+            # Restore the original $env:path
+            $env:path = $old_path
+        }
     }
-    Optimize-PowershellAssemblies
+    
+    # Optimize PowerShell assemblies:
+    Optimize-Assemblies -assemblyFilter "Microsoft.PowerShell."
```

Comment:
```suggestion
    Optimize-Assemblies -AssemblyFilter "Microsoft.PowerShell."
```

In this example, why set the filter again to the same value as the default?

Reply:
Again this is a case of an example. Effectively I know that this will result in the same thing happening. 
However, what it also shows is the usecase of the filter and it allow modification if you desire. 
Though I'd say it's a nice to have and not a need to have. 
In my experience, it is best practice to have any passable value have a default value.